### PR TITLE
Adding BasicAuth 

### PR DESCRIPTION
**Pull request by bikrdotnet**
This pull request includes a

- [ ] Bug fix
- [ x ] New feature
- [ ] Translation

The following changes were made

- Added BasicAuth 
     Default username and password: pokemon / pokemon 

- Added  Flask-BasicAuth==0.2.0 to requirements.txt

### CODE REVIEW

File: runserver.py [hunk 1]
```diff
@@ -58,6 +60,13 @@ def start_locator_thread(args):
         insert_mock_data(args.location, 6)
 
     app = Pogom(__name__)
+    
+    app.config['BASIC_AUTH_USERNAME'] = 'pokemon'
+    app.config['BASIC_AUTH_PASSWORD'] = 'pokemon'
+
+    basic_auth = BasicAuth(app)
+    app.config['BASIC_AUTH_FORCE'] = True
```
Comment (ellisio) [inline runserver.py]: This is enabling this by default, isn't it? Probably should set this to `False` and then add a flag on the args.

**Comment by Ashex1** [comment]
Seems if authentication is going to be implemented that this should be configurable either as an argument or from a file. Hardcoded credentials basically prevents people from using the project and pulling changes in easily.

**Comment by talentedmrripley** [comment]
Perhaps it could get the credentials from credentials.json.  And I agree it should default to false with a flag to enable.

**Comment by bikrdotnet** [comment]
Thanks for the comments.  Going to dig in and see how the parameters work and relate to the py.

**Comment by bikrdotnet** [comment]
New PR Coming.

**Pull request closed without merging**